bug(StoreForward): Add missing retrieval of DB credentials from Vault

lenny-intel opened the pull request:
Add the ability to pull the database credentials from Vault when security is enabled.
This is required for Store and Forward when security is enabled.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/edgexfoundry/app-functions-sdk-go/blob/master/.github/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:


## What is the current behavior?
Currently plain-text DB credentials are always used.

Issue Number: #243 

## What is the new behavior?
Now when security is enable (default) the DB credentials are pulled from Vault 

The Mongo DB connection code had to be tweaked to avoid using the mongo-driver url parser which doesn't allow `/` in the password, which can occur with the current dynamic password generation.

The code in internal/security was copied from edgex-go with minor tweaks for running for the SDk, not the bootstrap loader.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

michaelestrin commented:
@lenny-intel -- please remove @michaelestrin and @brandonforster  as reviewers and add @AnthonyMBonafide (who implemented this as a common bootstrap in core).

lenny-intel commented:
@michaelestrin , I have removed you and added @AnthonyMBonafide. I specifically added @brandonforster for the changes I made to his code for the mongo connection.

lenny-intel commented:
@AnthonyMBonafide, 

FYI @jim-wang-intel and I validated this PR with nightly builds compose files. Security and Non Security.
